### DISCUSSION TITLE
How can i integrate prometheus exporter with falcon?

### DISCUSSION THREAD

**Discussion by david-gang** [falconry/falcon#2161] [Q&A] [answered]
Hi,
I need your help in integrating prometheus.
I want to use prometheus together with falcon.

I know that [falcon prometheus exporter](https://gitlab.com/shdh/falcon-prometheus/-/blob/master/falcon_prometheus/middleware.py) exists but i don't like it too much as for example it does not serve openmetrics seamless, so it would make sense to use a dispatcher middleware like in flask https://github.com/prometheus/client_python/tree/master#flask
or to have a mount of another wsgi app like in fastapi https://github.com/prometheus/client_python/tree/master#fastapi--gunicorn

I don't find any way how to make this easily in falcon, so i can copy the code, or i am stuck here. what can i do ?

The best solution i came app was the following:

```
app  = falcon.App()
prometheusApp = make_wsgi_app()
def application(environ, start_response):
    path = environ.get('PATH_INFO')
    if path == '/metrics':
        return prometheusApp(environ, start_response)
    else:
        return app(environ, start_response)
```

This is based on https://falcon.readthedocs.io/en/stable/user/faq.html#how-do-i-split-requests-between-my-original-app-and-the-part-i-migrated-to-falcon

but i feel somehow that this is ugly and there should be some more "falconic way to do this stuff"

**Comment by vytas7** [comment, marked as answer]
Hi @david-gang!
I don't think this is ugly _per se_, but it is still going an extra mile. At the time of writing this is the recommended workaround.

As mentioned on Gitter, there are improvements planned for the 4.x series both regarding better instrumentation (although we will likely only support the OpenTelemetry protocol in the first iteration), and mounting other applications.

See also:

- #1828
- #1950
- #2162
